### ISSUE TITLE
Cannot use 'in' operator to search for 'imageSmoothingEnabled' in null

### ISSUE THREAD

**Issue by user753** [photonstorm/phaser-ce#487]
* Phaser version(s): 2.10

I cannot recreate this error and the game works fine for most users, but some users experience this error:
```
Cannot use 'in' operator to search for 'imageSmoothingEnabled' in null
```
And it occurs only in Chromium browsers.
Game settings
```
    {
      width: 2080,
      height: 1280,
      renderer: Phaser.AUTO,
      parent: "game",
      transparent: true,
      disableVisibilityChange: true,
      failIfMajorPerformanceCaveat: true,
    }
```
```
Object.getSmoothingPrefix at line 3:507589
new s.CanvasRenderer at line 3:192693
i.Game.setUpRenderer at line 3:289390
i.Game.boot at line 3:286307
i.Device._readyCheck at line 3:497824
```
It looks like CanvasRenderingContext2D is null in some cases.
What could be the reason?

**Comment by samme** [comment]
That could only be because context creation failed (thus [CanvasRenderer#context](https://photonstorm.github.io/phaser-ce/PIXI.CanvasRenderer.html#context) is null).

I can make the CanvasRenderer fail earlier, but there's no preventing it. It may be those users have Canvas disabled somehow.

**Comment by user753** [comment]
@samme Could you try to use WebGL in this case? If canvas isn't available for some reason I think it's a reasonable approach.
And throw the error if both method cannot be used.

**Comment by samme** [comment]
If you're using renderer AUTO then WEBGL was tried first, before CANVAS.

**Comment by user753** [comment]
But if I use renderer CANVAS? Maybe it would be better if game worked with WEBGL in this case?

**Comment by samme** [comment]
If you're using AUTO already and you're seeing this error, it means that the WebGL context was tried first and failed and then the Canvas 2d context was tried and also failed.

You might try using `failIfMajorPerformanceCaveat: false` instead.

**Comment by user753** [comment]
@samme No, I mean if I use `renderer: Phaser.CANVAS` and Phaser cannot create canvas context shouldn't Phaser try to use WEBGL instead?

**Comment by samme** [comment]
No, only AUTO does that.

**Comment by samme** [comment]
Do try `failIfMajorPerformanceCaveat: false` though. It's quite possible some of those Chromium browsers are obeying that and giving up early.

**Comment by samme** [comment]
Although that doesn't explain why CANVAS would fail as well. It could be that Chromium has disabled all canvas contexts on the particular device. I think you'll have to find one of these users to figure out what's going on.
